infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/LogicalGroupLogEntryReader.java

```diff
@@ -12,10 +34,99 @@
  * with the exception that it will read from a different transactional stream for log entry sync (one that is
  * specific for this model).
  */
+@Slf4j
 public class LogicalGroupLogEntryReader extends BaseLogEntryReader {
 
+    public static final UUID CLIENT_CONFIG_TABLE_ID = CorfuRuntime.getStreamID(
+            TableRegistry.getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, LR_MODEL_METADATA_TABLE_NAME));
+
+    private final ISerializer protobufSerializer;
+
+
     public LogicalGroupLogEntryReader(CorfuRuntime runtime, LogReplicationSession session,
                                       LogReplicationContext replicationContext) {
         super(runtime, session, replicationContext);
+        protobufSerializer = replicationContext.getProtobufSerializer();
+    }
+
+    /**
+     * Verify the transaction entry is valid. For LOGICAL_GROUP case, it will check:
+     * (1) If current session is impacted by group destinations change
+     * (2) If the opaque entry contains newly opened streams to replicate
+     * <p>
+     * Notice that a transaction stream entry can be fully or partially replicated,
+     * i.e., if only a subset of streams in the transaction entry are part of the streams
+     * to replicate, the transaction entry will be partially replicated,
+     * avoiding replication of the other streams present in the transaction.
+     *
+     * @param entry transaction stream opaque entry
+     * @return true, if the transaction entry has any valid stream to replicate.
+     * false, otherwise.
+     */
+    @Override
+    boolean isValidTransactionEntry(@NonNull OpaqueEntry entry) {
+        Set<UUID> txEntryStreamIds = new HashSet<>(entry.getEntries().keySet());
+
+        if (txEntryStreamIds.contains(CLIENT_CONFIG_TABLE_ID) &&
+                isCurrentSessionImpacted(entry.getEntries().get(CLIENT_CONFIG_TABLE_ID))) {
+            log.info("Group destination change detected, log entry sync will be stopped and a snapshot sync " +
```

Comment:
Since this is not a forced snapshot sync, the log message can be updated accordingly

Reply:
yes, updated